### PR TITLE
Added SYSTEM_CHAT and DISGUISED_CHAT to translation packet adapter

### PR DESCRIPTION
**Pull request by paaulhier**

### CODE REVIEW

File: i18n-paper/src/main/java/de/helixdevs/i18n/paper/protocollib/ChatAdapter.java [hunk 1]
```diff
@@ -2,16 +2,24 @@
...
+                PacketType.Play.Server.CHAT,
+                PacketType.Play.Server.SYSTEM_CHAT,
+                PacketType.Play.Server.DISGUISED_CHAT
+        );
     }
 
     @Override
     public void onPacketSending(PacketEvent event) {
-        handleChatComponent(event.getPlayer(), event.getPacket(), 0);
+        var packet = event.getPacket();
```
Comment (mcmdev) [inline i18n-paper/src/main/java/de/helixdevs/i18n/paper/protocollib/ChatAdapter.java]: No var. Use typed variables

File: i18n-paper/src/main/java/de/helixdevs/i18n/paper/protocollib/util/ReflectiveAdventureComponentConverter.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
+package de.helixdevs.i18n.paper.protocollib.util;
+
+import com.comphenix.protocol.reflect.FuzzyReflection;
+import com.comphenix.protocol.reflect.accessors.Accessors;
+import com.comphenix.protocol.reflect.accessors.MethodAccessor;
+import com.comphenix.protocol.utility.MinecraftReflection;
+import com.comphenix.protocol.wrappers.WrappedChatComponent;
+
+public class ReflectiveAdventureComponentConverter {
```
Comment (mcmdev) [inline i18n-paper/src/main/java/de/helixdevs/i18n/paper/protocollib/util/ReflectiveAdventureComponentConverter.java]: Use https://github.com/dmulloy2/ProtocolLib/blob/master/src/main/java/com/comphenix/protocol/wrappers/AdventureComponentConverter.java#

File: i18n-paper/src/main/java/de/helixdevs/i18n/paper/protocollib/ITranslationAdapter.java [hunk 3]
```diff
@@ -86,7 +109,11 @@ default WrappedChatComponent translate(Player player, WrappedChatComponent wrapp
         if (wrappedChatComponent == null) {
             return null;
         }
-        return WrappedChatComponent.fromJson(gsonComponentSerializer.serialize(GlobalTranslator.render(gsonComponentSerializer.deserialize(wrappedChatComponent.getJson()), player.locale())));
+        var component = GlobalTranslator.render(gsonComponentSerializer.deserialize(wrappedChatComponent.getJson()), player.locale());
+        Bukkit.getLogger().info("Translation:" + gsonComponentSerializer.serialize(component));
```
Comment (mcmdev) [inline i18n-paper/src/main/java/de/helixdevs/i18n/paper/protocollib/ITranslationAdapter.java]: Remove this debug

File: i18n-paper/src/main/java/de/helixdevs/i18n/paper/protocollib/ITranslationAdapter.java [hunk 4]
```diff
@@ -86,7 +109,11 @@ default WrappedChatComponent translate(Player player, WrappedChatComponent wrapp
         if (wrappedChatComponent == null) {
             return null;
         }
-        return WrappedChatComponent.fromJson(gsonComponentSerializer.serialize(GlobalTranslator.render(gsonComponentSerializer.deserialize(wrappedChatComponent.getJson()), player.locale())));
+        var component = GlobalTranslator.render(gsonComponentSerializer.deserialize(wrappedChatComponent.getJson()), player.locale());
```
Comment (mcmdev) [inline i18n-paper/src/main/java/de/helixdevs/i18n/paper/protocollib/ITranslationAdapter.java]: Same thing with var here

**Review by mcmdev — CHANGES_REQUESTED** [review]

File: i18n-paper/src/main/java/de/helixdevs/i18n/paper/protocollib/ITranslationAdapter.java (hunk 4 above)
Comment (paaulhier) [reply i18n-paper/src/main/java/de/helixdevs/i18n/paper/protocollib/ITranslationAdapter.java]: done

**Comment by mcmdev** [comment]
Is this tested?

**Comment by paaulhier** [comment]
it is

**Pull request merged**